core/lib/Cardano/Address/Errors.hs

```diff
@@ -0,0 +1,91 @@
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+-- |
+-- Copyright: © 2020 IOHK
+-- License: Apache-2.0
+
+module Cardano.Address.Errors
+    ( -- $overview
+
+      -- * AddrError
+      ShelleyAddrError(..)
+    , ByronAddrError(..)
+    , IcarusAddrError(..)
+    , JormungandrAddrError(..)
+
+      -- * pretty printers
+    , prettyShError
+    , prettyBrError
+    , prettyIcError
+    , prettyJoError
+  ) where
+
+import Prelude
+
+import Control.Exception
+    ( Exception )
+import Fmt
+    ( format )
+
+
+-- $overview
+--
+-- This module provides error/exception types for Addresses.
+
+data ShelleyAddrError
+    = ShUnknownAddrType
+    | ShWrongInputSize Int -- the actual size
+    | ShPtrRetrieveError String -- human readable error of underlying operation
+
+instance Exception ShelleyAddrError
+
+prettyShError :: ShelleyAddrError -> String
+prettyShError ShUnknownAddrType = "Unknown address type"
+prettyShError (ShWrongInputSize i) = format "Wrong input size of {}" i
+prettyShError (ShPtrRetrieveError s) = format "Failed to retrieve pointer (underlying errors was: {})" s
+
+instance Show ShelleyAddrError where
+  show = prettyShError
```

Comment:
:upside_down_face: `Show` instances should not be written by hand; `fmt` provides a `Buildable` instance for human-readable formatting! The purpose of `Show` is for debug only.